apps/server/src/services/institutionResolver.ts

```diff
@@ -88,18 +91,61 @@ export async function resolveInstitutionProvider(
 
 function getAvailableProviders(
   institution: CachedInstitution,
+  jobType: JobType,
   supportedProviders?: Provider[]
 ): Provider[] {
   const providers = []
-  if (supportedProviders.includes('mx') && institution.mx.id != null) {
+  if (
+    supportedProviders.includes('mx') &&
+    institution.mx.id != null &&
+    providerSupportsJobType(institution.mx, jobType)
+  ) {
     providers.push('mx')
   }
   if (
     supportedProviders.includes('sophtron') &&
-    institution.sophtron.id != null
+    institution.sophtron.id != null &&
+    providerSupportsJobType(institution.sophtron, jobType)
   ) {
     providers.push('sophtron')
   }
 
   return providers as Provider[]
 }
+
+function providerSupportsJobType(
+  institutionAttributes: InstitutionProvider | undefined,
+  jobType: JobType
+): boolean {
+  if (institutionAttributes == null) {
```

Comment:
I feel like a better check here would be: 

`if (!institutionAttributes) {`

Checking for null is different than a variable or parameter that hasn't been passed in, which generally would be `undefined`. 

If you know that the possible values of `institutionAttributes` will sometimes be `null`, then the check you have will work as expected.

Reply:
Actually, because you are doing a loose comparison (`==`), it will probably work as expected.

Reply:
Agreed on the !